### ISSUE TITLE
HTML mode: "&nbsp;" is set often for normal whitespaces and leads to problems

### ISSUE THREAD

**Issue by q2apro** [samclarke/SCEditor#646]
Here is an example post: 

![image](https://user-images.githubusercontent.com/6764548/35562409-efda2c18-05bb-11e8-9866-fd8bac619dc0.png)

The HTML created by SCEditor: 

`<p>Falls ihr noch keinen Paypal.me-Link habt, könnt ihr diesen hier bekommen:&nbsp;<a href="https://www.paypal.com/paypalme/grab" rel="nofollow" target="_blank">https://www.paypal.com/paypalme/grab</a> - Selbstverständlich braucht ihr vorher ein Paypal-Konto.</p>`

As we can see there was an HTML whitespace set `bekommen:&nbsp;` which connects the text with the anchor link, creating a long text line that the Chrome browser puts onto the next line.

Can we avoid the `&nbsp;` if they are single? Or what other solution could there be?

**Comment by q2apro** [comment]
Other examples: 

`<p>
	Damit ihr als&nbsp;Mitglied dieses Feature aktivieren könnt, müsst ihr&nbsp;mindestens <strong>1000 Punkte</strong>&nbsp;auf der Mathelounge/Nanolounge (getrennte Punktesysteme) haben.&nbsp;
</p>`

Especially the whitespace at the end of the line makes no sense :)

**Comment by samclarke** [comment]
You're right the editor should normalise this. It should preserve user inserted non-breaking spaces but remove automatically inserted ones when they're not next to another space.

**Comment by samclarke** [comment]
I've just checked and at least in Chrome this seems to already be mostly handled. For some reason it uses non-breaking spaces around links but can't get it to happen between words. What browser were you using to get the second example?

**Comment by q2apro** [comment]
For all examples I was using Chrome Version 63.0.3239.132 (64 bit).

**Comment by q2apro** [comment]
Can I suggest to add `&nbsp;` only if the spaces has been hit two times? Then a `some text: <a href="...` would not be parsed to `some text:&nbsp;<a href="...`

**Comment by q2apro** [comment]
Can we get the status from "Minor bug" to "Bug"?

I just checked the posts of the last four weeks in my production enviroment. Result 32407 occurences of `&nbsp;<`.
